Expensify/App
# Emoji Picker Menu Navigation

### Details
Added the ability to navigate through the emoji picker menu with arrow keys, and to send the highlighted emoji with [Enter]. It should function the same as the emoji picker in Slack.
- the highlighted emoji should never be cutoff by the window
- the first emoji is highlighted upon searching emojis
- hovering over an emoji with your cursor will also change the currently highlighted emoji
- if no emoji is highlighted, pressing an arrow key will highlight the first emoji
- the cursor in the search input is not affected by arrow key presses that change the highlighted emoji

### Fixed Issues
Fixes https://github.com/Expensify/Expensify.cash/issues/2450

### Tests

I tested this feature on Web and Desktop since those are the only two places it can actually be used, but I also tested it on Mobile Web to ensure that the touchscreen check works and that the Emoji Picker Menu doesn't crash.

1. Opened the emoji picker. Scroll down till you can see the end of on category and the start of another.
2. Use your mouse to hover over an emoji near this boundary.
3. Use the up/down arrow keys to move the highlight around this border. Verified that it skips over the header.
4. Move the highlight at the top and bottom edges of the window with the arrow keys. Verified that the menu scrolls to keep the highlighted emoji in the window.
5. Type in search terms to filter by. Verified that the first search result is highlighted. Verified that nothing breaks when you try to go out of bounds. Verified that you can move the cursor using the appropriate arrow keys when they would move the highlight out of bounds.
6. Pressed [Enter]. Verified that the highlighted emoji was inputted into the compose box.
7. Repeat steps 1-6 two more times to verify that event listeners are removed and added correctly.

### QA Steps
Identical to the tests above. See videos for examples.

### Tested On


### Screenshots

#### Web
https://user-images.githubusercontent.com/31285285/115820241-0048f200-a433-11eb-9dd8-d6c274158365.mp4

#### Desktop
https://user-images.githubusercontent.com/31285285/115819992-8ca6e500-a432-11eb-9322-4cabf9f107f3.mp4

cc @stitesExpensify
